src/tabbar.h

```diff
@@ -10,6 +10,24 @@
 #include "fullscreenwindow.h"
 #include <QMouseEvent>
 #include <QWebEngineFullScreenRequest>
+#include <QJsonObject>
+
+struct BookInfo 
+{
+    QString name = "N/A";
+    QString path = "N/A";
+
+    BookInfo() {};
+    BookInfo(const QString &name, const QString &path)
+        : name{name}, path{path} {};
+    BookInfo(const QJsonObject &json)
+        : name{json.value("name").toString()},
+          path{json.value("path").toString()} {};
+
+    QJsonObject toJson() const { return {{"name", name}, {"path", path}}; };
+};
```

Comment:
When you were referring to the complicated solution I thought that you meant the last version (that had some issues because of directory monitoring) before we decided to split the PR. Sorry.

Reply:
Since we cannot change the behavior of directory monitoring, we can:
1. simply save all the books that we have seen in updateFromDir, no matter if we will be removing them.
2. only remove a saved book when all traces such as opened tab and bookmarks of that book is removed.